{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting PyAthena\n",
      "\u001b[?25l  Downloading https://files.pythonhosted.org/packages/c5/ff/e20073f4be9592f2daf9c3f06e7e560221bfec4c9cbb06b1eeb6266da6f4/PyAthena-1.4.6-py2.py3-none-any.whl (49kB)\n",
      "\u001b[K    100% |████████████████████████████████| 51kB 2.2MB/s ta 0:00:01\n",
      "\u001b[?25hCollecting future (from PyAthena)\n",
      "\u001b[?25l  Downloading https://files.pythonhosted.org/packages/90/52/e20466b85000a181e1e144fd8305caf2cf475e2f9674e797b222f8105f5f/future-0.17.1.tar.gz (829kB)\n",
      "\u001b[K    100% |████████████████████████████████| 829kB 12.2MB/s ta 0:00:01\n",
      "\u001b[?25hRequirement already satisfied: botocore>=1.5.52 in /home/ec2-user/anaconda3/envs/python2/lib/python2.7/site-packages (from PyAthena) (1.12.97)\n",
      "Requirement already satisfied: futures; python_version == \"2.7\" in /home/ec2-user/anaconda3/envs/python2/lib/python2.7/site-packages (from PyAthena) (3.2.0)\n",
      "Collecting tenacity>=4.1.0 (from PyAthena)\n",
      "  Downloading https://files.pythonhosted.org/packages/aa/38/16856e4df287ad7a5fe8602d57f04955d77b8f95b7e5302517a4b3df619a/tenacity-5.0.3-py2.py3-none-any.whl\n",
      "Requirement already satisfied: boto3>=1.4.4 in /home/ec2-user/anaconda3/envs/python2/lib/python2.7/site-packages (from PyAthena) (1.9.97)\n",
      "Requirement already satisfied: jmespath<1.0.0,>=0.7.1 in /home/ec2-user/anaconda3/envs/python2/lib/python2.7/site-packages (from botocore>=1.5.52->PyAthena) (0.9.3)\n",
      "Requirement already satisfied: docutils>=0.10 in /home/ec2-user/anaconda3/envs/python2/lib/python2.7/site-packages (from botocore>=1.5.52->PyAthena) (0.14)\n",
      "Requirement already satisfied: python-dateutil<3.0.0,>=2.1; python_version >= \"2.7\" in /home/ec2-user/anaconda3/envs/python2/lib/python2.7/site-packages (from botocore>=1.5.52->PyAthena) (2.7.3)\n",
      "Requirement already satisfied: urllib3<1.25,>=1.20; python_version == \"2.7\" in /home/ec2-user/anaconda3/envs/python2/lib/python2.7/site-packages (from botocore>=1.5.52->PyAthena) (1.22)\n",
      "Collecting monotonic>=0.6; python_version == \"2.7\" (from tenacity>=4.1.0->PyAthena)\n",
      "  Downloading https://files.pythonhosted.org/packages/ac/aa/063eca6a416f397bd99552c534c6d11d57f58f2e94c14780f3bbf818c4cf/monotonic-1.5-py2.py3-none-any.whl\n",
      "Requirement already satisfied: six>=1.9.0 in /home/ec2-user/anaconda3/envs/python2/lib/python2.7/site-packages (from tenacity>=4.1.0->PyAthena) (1.11.0)\n",
      "Requirement already satisfied: s3transfer<0.3.0,>=0.2.0 in /home/ec2-user/anaconda3/envs/python2/lib/python2.7/site-packages (from boto3>=1.4.4->PyAthena) (0.2.0)\n",
      "Building wheels for collected packages: future\n",
      "  Running setup.py bdist_wheel for future ... \u001b[?25ldone\n",
      "\u001b[?25h  Stored in directory: /home/ec2-user/.cache/pip/wheels/0c/61/d2/d6b7317325828fbb39ee6ad559dbe4664d0896da4721bf379e\n",
      "Successfully built future\n",
      "Installing collected packages: future, monotonic, tenacity, PyAthena\n",
      "Successfully installed PyAthena-1.4.6 future-0.17.1 monotonic-1.5 tenacity-5.0.3\n",
      "\u001b[33mYou are using pip version 10.0.1, however version 19.0.3 is available.\n",
      "You should consider upgrading via the 'pip install --upgrade pip' command.\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "import sys\n",
    "!{sys.executable} -m pip install PyAthena"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "from pyathena import connect"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [],
   "source": [
    "conn = connect(s3_staging_dir='s3://aws-athena-query-results-417383318849-us-east-2/',\n",
    "               region_name='us-east-2')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<pyathena.connection.Connection at 0x7f590f9a8f10>"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "conn"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_sql(\"SELECT * FROM default.cr22_variant limit 8;\", conn)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>variant</th>\n",
       "      <th>variantcallingannotations</th>\n",
       "      <th>sampleid</th>\n",
       "      <th>sampledescription</th>\n",
       "      <th>processingdescription</th>\n",
       "      <th>alleles</th>\n",
       "      <th>expectedalleledosage</th>\n",
       "      <th>referencereaddepth</th>\n",
       "      <th>alternatereaddepth</th>\n",
       "      <th>readdepth</th>\n",
       "      <th>minreaddepth</th>\n",
       "      <th>genotypequality</th>\n",
       "      <th>genotypelikelihoods</th>\n",
       "      <th>nonreferencelikelihoods</th>\n",
       "      <th>strandbiascomponents</th>\n",
       "      <th>splitfrommultiallelic</th>\n",
       "      <th>isphased</th>\n",
       "      <th>phasesetid</th>\n",
       "      <th>phasequality</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>{varianterrorprobability=100, contig={contigna...</td>\n",
       "      <td>{variantispassing=true, variantfilters=[], dow...</td>\n",
       "      <td>HG00096</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>[Ref, Ref]</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>[]</td>\n",
       "      <td>[]</td>\n",
       "      <td>[]</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>{varianterrorprobability=100, contig={contigna...</td>\n",
       "      <td>{variantispassing=true, variantfilters=[], dow...</td>\n",
       "      <td>HG00097</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>[Ref, Ref]</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>[]</td>\n",
       "      <td>[]</td>\n",
       "      <td>[]</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>{varianterrorprobability=100, contig={contigna...</td>\n",
       "      <td>{variantispassing=true, variantfilters=[], dow...</td>\n",
       "      <td>HG00099</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>[Ref, Ref]</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>[]</td>\n",
       "      <td>[]</td>\n",
       "      <td>[]</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>{varianterrorprobability=100, contig={contigna...</td>\n",
       "      <td>{variantispassing=true, variantfilters=[], dow...</td>\n",
       "      <td>HG00100</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>[Ref, Ref]</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>[]</td>\n",
       "      <td>[]</td>\n",
       "      <td>[]</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>{varianterrorprobability=100, contig={contigna...</td>\n",
       "      <td>{variantispassing=true, variantfilters=[], dow...</td>\n",
       "      <td>HG00101</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>[Ref, Ref]</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>[]</td>\n",
       "      <td>[]</td>\n",
       "      <td>[]</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>{varianterrorprobability=100, contig={contigna...</td>\n",
       "      <td>{variantispassing=true, variantfilters=[], dow...</td>\n",
       "      <td>HG00102</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>[Ref, Ref]</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>[]</td>\n",
       "      <td>[]</td>\n",
       "      <td>[]</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>{varianterrorprobability=100, contig={contigna...</td>\n",
       "      <td>{variantispassing=true, variantfilters=[], dow...</td>\n",
       "      <td>HG00103</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>[Ref, Ref]</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>[]</td>\n",
       "      <td>[]</td>\n",
       "      <td>[]</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>{varianterrorprobability=100, contig={contigna...</td>\n",
       "      <td>{variantispassing=true, variantfilters=[], dow...</td>\n",
       "      <td>HG00105</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>[Ref, Ref]</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>[]</td>\n",
       "      <td>[]</td>\n",
       "      <td>[]</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                             variant  \\\n",
       "0  {varianterrorprobability=100, contig={contigna...   \n",
       "1  {varianterrorprobability=100, contig={contigna...   \n",
       "2  {varianterrorprobability=100, contig={contigna...   \n",
       "3  {varianterrorprobability=100, contig={contigna...   \n",
       "4  {varianterrorprobability=100, contig={contigna...   \n",
       "5  {varianterrorprobability=100, contig={contigna...   \n",
       "6  {varianterrorprobability=100, contig={contigna...   \n",
       "7  {varianterrorprobability=100, contig={contigna...   \n",
       "\n",
       "                           variantcallingannotations sampleid  \\\n",
       "0  {variantispassing=true, variantfilters=[], dow...  HG00096   \n",
       "1  {variantispassing=true, variantfilters=[], dow...  HG00097   \n",
       "2  {variantispassing=true, variantfilters=[], dow...  HG00099   \n",
       "3  {variantispassing=true, variantfilters=[], dow...  HG00100   \n",
       "4  {variantispassing=true, variantfilters=[], dow...  HG00101   \n",
       "5  {variantispassing=true, variantfilters=[], dow...  HG00102   \n",
       "6  {variantispassing=true, variantfilters=[], dow...  HG00103   \n",
       "7  {variantispassing=true, variantfilters=[], dow...  HG00105   \n",
       "\n",
       "  sampledescription processingdescription     alleles expectedalleledosage  \\\n",
       "0              None                  None  [Ref, Ref]                 None   \n",
       "1              None                  None  [Ref, Ref]                 None   \n",
       "2              None                  None  [Ref, Ref]                 None   \n",
       "3              None                  None  [Ref, Ref]                 None   \n",
       "4              None                  None  [Ref, Ref]                 None   \n",
       "5              None                  None  [Ref, Ref]                 None   \n",
       "6              None                  None  [Ref, Ref]                 None   \n",
       "7              None                  None  [Ref, Ref]                 None   \n",
       "\n",
       "  referencereaddepth alternatereaddepth readdepth minreaddepth  \\\n",
       "0               None               None      None         None   \n",
       "1               None               None      None         None   \n",
       "2               None               None      None         None   \n",
       "3               None               None      None         None   \n",
       "4               None               None      None         None   \n",
       "5               None               None      None         None   \n",
       "6               None               None      None         None   \n",
       "7               None               None      None         None   \n",
       "\n",
       "  genotypequality genotypelikelihoods nonreferencelikelihoods  \\\n",
       "0            None                  []                      []   \n",
       "1            None                  []                      []   \n",
       "2            None                  []                      []   \n",
       "3            None                  []                      []   \n",
       "4            None                  []                      []   \n",
       "5            None                  []                      []   \n",
       "6            None                  []                      []   \n",
       "7            None                  []                      []   \n",
       "\n",
       "  strandbiascomponents  splitfrommultiallelic  isphased phasesetid  \\\n",
       "0                   []                  False      True       None   \n",
       "1                   []                  False      True       None   \n",
       "2                   []                  False      True       None   \n",
       "3                   []                  False      True       None   \n",
       "4                   []                  False      True       None   \n",
       "5                   []                  False      True       None   \n",
       "6                   []                  False      True       None   \n",
       "7                   []                  False      True       None   \n",
       "\n",
       "  phasequality  \n",
       "0         None  \n",
       "1         None  \n",
       "2         None  \n",
       "3         None  \n",
       "4         None  \n",
       "5         None  \n",
       "6         None  \n",
       "7         None  "
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [],
   "source": [
    "import sklearn"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "conda_python2",
   "language": "python",
   "name": "conda_python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.15"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
